File: TeamUp.Server/Models/Game.cs
using System.ComponentModel.DataAnnotations.Schema;
using TeamUp.Server.Utils;

namespace TeamUp.Server.Models;

public class Game
{
    public int Id { get; set; }
    public DateTime Date { get; set; }
    //public TimeSpan Time { get; set; }
    public string Location { get; set; }
    public int ScoreTeam1 { get; set; }
    public int ScoreTeam2 { get; set; }
    public Team? Team1 { get; set; }
    public Team? Team2 { get; set; }
    public List<Player> Players = new();

    public string Status { get; set; } = "Scheduled";

    public Game(DateTime date, string location)
    {
        Date = date; 
        Location = location;
    }
}

File: teamup.client/src/app/models/Game.ts
import { Player } from "./Player";
import { Team } from "./Team";

export class Game {
   id: number;
   date: Date;
   location: string;
   scoreTeam1?: number;
   scoreTeam2?: number;
   team1Id?: number;
   team1?: Team;
   team2Id?: number;
   team2?: Team;
   status: string;
   players: Player[];
}
